Laiban/Modules/Recreation/Sources/Services/RecreationService.swift

```diff
@@ -3,16 +3,163 @@
 //
 
 import Foundation
-
+import SwiftUI
 import Combine
 
-public class RecreationService: ObservableObject, LBDashboardItem {
+public typealias RecreationServiceType = [Recreation]
+public typealias RecreationStorageService = CodableLocalJSONService<RecreationServiceType>
+
+
+public enum InventoryType: String, CaseIterable, Identifiable {
+    case misc = "Diverse", animals = "Djur", songs = "Sånger"
+    public var id: Self { self }
+}
+
+enum RecreationType {
+    case Inventory
+    case Activity
+}
+
+
+public class RecreationService: CTS<RecreationServiceType,RecreationStorageService>, LBDashboardItem, LBAdminService {
+    
+    private let defaultIndexForRecreationObject = 0
+    
+    
+    public var id: String = "RecreationService"
+    public var listOrderPriority: Int = 15
+    public var listViewSection: LBAdminListViewSection = .content
+    
     public let viewIdentity: LBViewIdentity = .recreation
     public var isAvailablePublisher: AnyPublisher<Bool, Never> {
         $isAvailable.eraseToAnyPublisher()
     }
     @Published public private(set) var isAvailable: Bool = true
-    public init() {
+    
+    public func adminView() -> AnyView {
+        AnyView(RecreationAdminView(service: self))
+    }
+    
+    public var cancellables = Set<AnyCancellable>()
+    
+    
+    public convenience init() {
+        self.init(
+            emptyValue: [],
+            storageOptions: .init(filename: "Recreation", foldername: "RecreationService", bundleFilename:"Recreation", bundle:.module)
+        )
+        
+        Task {
+            await self.load()
+        }
+       
+    }
+    
+    func getRecreation() -> Recreation {
```

Comment:
Maybe rename to `getActiveRecreations()`? Maybe also a computed var to match the suggestion above:
```
public var activeRecreations: Recreation {
...
}
```